relayer/chain/ethereum/writer.go

```diff
@@ -134,3 +197,118 @@ func (wr *Writer) Write(ctx context.Context, msg *chain.SubstrateOutboundMessage
 
 	return nil
 }
+
+func (wr *Writer) WriteNewSignatureCommitment(ctx context.Context, item *store.BeefyItem, valIndex int) error {
+	beefy, err := item.ToBeefy()
+	if err != nil {
+		wr.log.WithError(err).Error("Error converting database item to Beefy")
+	}
+
+	msg, err := beefy.BuildNewSignatureCommitmentMessage(valIndex)
+	if err != nil {
+		return err
+	}
+
+	// inSet, err := wr.CheckValidatorInSet(ctx, msg.ValidatorPublicKey, msg.ValidatorPosition, msg.ValidatorPublicKeyMerkleProof)
+	// if err != nil {
+	// 	return err
+	// }
+	// if !inSet {
+	// 	return fmt.Errorf("validator address merkle proof failed verification")
+	// }
+
+	contract := wr.polkadotRelayChainBridge
+	if contract == nil {
+		return fmt.Errorf("Unknown contract")
+	}
+
+	blockNumber, err := wr.conn.client.BlockNumber(ctx)
+	if err != nil {
+		return err
+	}
+
+	options := bind.TransactOpts{
+		From:     wr.conn.GetKeyPair().CommonAddress(),
+		Signer:   wr.signerFn,
+		Context:  ctx,
+		GasLimit: 5000000,
+	}
+
+	tx, err := contract.NewSignatureCommitment(&options, msg.Payload,
+		msg.ValidatorClaimsBitfield, msg.ValidatorSignatureCommitment,
+		msg.ValidatorPosition, msg.ValidatorPublicKey, msg.ValidatorPublicKeyMerkleProof)
+	if err != nil {
+		wr.log.WithError(err).Error("Failed to submit transaction")
+		return err
+	}
+
+	wr.log.WithFields(logrus.Fields{
+		"txHash":      tx.Hash().Hex(),
+		"blockNumber": blockNumber,
```

Comment:
This blockNumber will not be from the block which includes the tx, which might be misleading. What is the intention here?

Reply:
Good catch. Removed `blockNumber` from the log in https://github.com/Snowfork/polkadot-ethereum/pull/314/commits/b40cf71eb5ce4c11b59406eab49421bb2245d5d3